includes/reports/reports-functions.php

```diff
@@ -384,73 +384,86 @@ function get_filter_value( $filter ) {
 		return $value;
 	}
 
-	// Look for filter in transients
-	$filter_key   = get_filter_key( $filter );
-	$filter_value = get_transient( $filter_key );
-
-	// Maybe use transient value
-	if ( false !== $filter_value ) {
-		$value = $filter_value;
-
-		// Maybe use dates defaults
-	} elseif ( 'dates' === $filter ) {
-
-		// Default to last 30 days for filter value.
-		$default = 'last_30_days';
-		$dates   = parse_dates_for_range( $default );
-		$value   = array(
-			'from'  => $dates['start']->format( 'Y-m-d' ),
-			'to'    => $dates['end']->format( 'Y-m-d' ),
-			'range' => $default,
-		);
-	}
+	switch ( $filter ) {
+		// Handle dates.
+		case 'dates':
+			if ( ! isset( $_GET['range'] ) ) {
+				$default = 'last_30_days';
+				$dates   = parse_dates_for_range( $default );
+				$value   = array(
+					'range' => $default,
+					'from'  => $dates['start']->format( 'Y-m-d' ),
+					'to'    => $dates['end']->format( 'Y-m-d' ),
+				);
+			} else {
+				$value = array(
+					'range' => sanitize_text_field( $_GET[ 'range' ] ),
+					'from'  => isset( $_GET['filter_from'] )
+						? sanitize_text_field( $_GET[ 'filter_from'] )
+						: '',
+					'to'    => isset( $_GET['filter_to'] )
+						? sanitize_text_field( $_GET[ 'filter_to'] )
+						: ''
+				);
+			}
 
-	return $value;
-}
+			break;
 
-/**
- * Sets the value of a given report filter.
- *
- * The filter will only be set if the filter is valid.
- *
- * @since 3.0
- *
- * @param string $filter Filter name.
- * @param mixed  $value  Filter value.
- */
-function set_filter_value( $filter, $value ) {
-	if ( validate_filter( $filter ) ) {
-		$filter_key = get_filter_key( $filter );
+		// Handle taxes.
+		case 'taxes':
+			$value = array();
 
-		set_transient( $filter_key, $value );
+			if ( isset( $_GET['exclude_taxes'] ) ) {
+				$value['exclude_taxes'] = true;
+			}
+
+			break;
+
+		// Handle default (direct from URL).
+		default:
+			$value = isset( $_GET[ $filter ] )
+				? sanitize_text_field( $_GET[ $filter ] )
+				: '';
+
+			/**
+			 * Filters the value of a report filter.
+			 *
+			 * @since 3.0
+			 *
+			 * @param string $value Report filter value.
+			 * @param string $filter Report filter.
+			 */
+			$value = apply_filters( 'edd_reports_get_filter_value', $value, $filter );
```

Comment:
Is this `apply_filters` within the `default` case? It seems like it should be at line 439 so that it applies to all?

Reply:
@robincornett My thinking here was an effort to reduce the potential for extension developers to accidentally break our built in filters. It can always be moved out at a later date but this gives us a bit of a buffer initially.